engines/python/setup/djl_python/request_output.py

```diff
@@ -0,0 +1,132 @@
+from dataclasses import field, dataclass
+from typing import List, Optional, Callable, Any, Dict, Union
+
+from djl_python.rolling_batch.rolling_batch import Token
+
+
+@dataclass
+class Sequence:
+    """Dataclass to store the generated tokens details of all the sequences in the request.
+    Attributes:
+        tokens: generated tokens of the sequence.
+        top_tokens: top tokens of the sequence.
+        cumulative_log_prob: cumulative log probability of the sequence.
+        finish_reason: finish reason of the sequence.
+        stop_reason: stop reason of the sequence.
+        generated_text: generated text of the sequence.
+    """
+    tokens: List[Token] = field(default_factory=lambda: [])
+    top_tokens: Optional[List[List[Token]]] = field(default_factory=lambda: [])
+    cumulative_log_prob: float = 0.0
+    finish_reason: str = False
+    stop_reason: Optional[str] = None
+    generated_text: str = ""
+
+    def set_next_token(self, token: Token):
+        self.tokens.append(token)
+        self.generated_text += token.text
+
+    def set_next_top_tokens(self, top_tokens: List[Token]):
+        self.top_tokens.append(top_tokens)
+
+    @property
+    def next_token(self):
```

Comment:
this should be called last token. And `@property` labe may not be correct. If I understand correctly, this will make a method to be permanent into element after you fetch once. For example:

```python
self.next_token() => 5
...
# it will be forever 5
self.next_token() => 5
```

It should reflect the iterator location and also give some boolean tells wether it is the end of current token list

Reply:
No, `@property` has getter and setters and it calls this getter each and everytime we get. But I understand property is more intended for read only properties. Can change this to getter method.